File: uniffi_bindgen/src/bindings/swift/templates/UInt64Helper.swift
fileprivate struct FfiConverterUInt64: FfiConverterPrimitive {
    typealias FfiType = UInt64
    typealias SwiftType = UInt64

    static func read(from buf: inout (data: Data, offset: Data.Index)) throws -> UInt64 {
        return try lift(readInt(&buf))
    }

    static func write(_ value: SwiftType, into buf: inout [UInt8]) {
        writeInt(&buf, lower(value))
    }
}

File: uniffi_bindgen/src/bindings/dart/templates/UInt64Helper.dart
typedef Uint64Ffi = Uint64;
typedef Uint64Lowered = Uint64;
typedef Uint64Lifted = int;
typedef Uint64DartFfi = int;

class FfiConverterUint64 {
    static Uint64Lowered lift(Uint64Lowered value) {
        return value;
    }

    static Uint64Lifted lower(Uint64Lifted value) {
        return value;
    }
}
